minerl/herobraine/env_spec.py

```diff
@@ -168,35 +168,28 @@ def is_from_folder(self, folder: str) -> bool:
 
     @abstractmethod
     def determine_success_from_rewards(self, rewards: list) -> bool:
-        # TODO: 
         raise NotImplementedError('subclasses must override determine_success_from_rewards()')
 
-    def _singlify(self, space: spaces.Dict):
-        if self.agent_count is 1:
-            return space.spaces[self.agent_names[0]]
-        else:
-            return space
-
     def create_observation_space(self):
-        return self._singlify(spaces.Dict({
+        return spaces.Dict({
             agent: spaces.Dict({
                 o.to_string(): o.space for o in self.observables
             }) for agent in self.agent_names
-        }))
+        })
 
     def create_action_space(self):
-        return self._singlify(spaces.Dict({
+        return spaces.Dict({
             agent: spaces.Dict({
                 a.to_string(): a.space for a in self.actionables
             }) for agent in self.agent_names
-        }))
+        })
 
     def create_monitor_space(self):
-        return self._singlify(spaces.Dict({
+        return spaces.Dict({

```

Comment:
Hmm, why are you removing the _singlify code? That should be breaking in the unit tests..

Reply:
The reason is that we want to be able to have an environment with a single agent while still using the multi-agent environment interface (which makes it much easier to write subsequent wrappers). However, _singlify only looks at self.agent_count, making it impossible to specify such an environment.

Perhaps the env_spec can have a separate parameter called `multi_agent_interface` (or something along those lines), which makes it possible to force a multi-agent interface in single-agent environments?

Reply:
What Joost is saying is exactly right: if we have environments that can return radically different formats depending on the number of agents, all the rest of our code will need to be special cased to handle that (I expect we'll often run sanity checks with a single agent), so it was simpler to just always return the multiagent format. I modified the unit tests that were broken by this accordingly.

If you prefer we can have a separate argument from agent_count handle this as Joost suggested.

Reply:
What if we just set agent_count=None and that specified this singular env? Or is that overloading behaviors too much

Reply:
Will do the agent_count=None version, seems reasonable to me.